In [3]:
import pandas as pd
import os
from dotenv import load_dotenv
load_dotenv()

df_matches = pd.read_csv("../data/raw/WorldCupMatches.csv")
df_cups = pd.read_csv("../data/raw/WorldCups.csv")

In [ ]:
print("Matches:", df_matches.head())
print("Cups:", df_cups.head())

df_matches.head()
df_cups.head()

API_KEY = os.getenv("FOOTBALL_API_KEY")

   Year      Country      Winner      Runners-Up    Third      Fourth  \
0  1930      Uruguay     Uruguay       Argentina      USA  Yugoslavia   
1  1934        Italy       Italy  Czechoslovakia  Germany     Austria   
2  1938       France       Italy         Hungary   Brazil      Sweden   
3  1950       Brazil     Uruguay          Brazil   Sweden       Spain   
4  1954  Switzerland  Germany FR         Hungary  Austria     Uruguay   

   GoalsScored  QualifiedTeams  MatchesPlayed Attendance  
0           70              13             18    590.549  
1           70              16             17    363.000  
2           84              15             18    375.700  
3           88              13             22  1.045.246  
4          140              16             26    768.607  


In [4]:
total_matches = df_matches.shape[0] #all the matches in all World Cup
missing_values = df_matches.isnull().sum() #empty values in the dataset
year_range = df_matches["Year"].max() - df_matches["Year"].min() #range of 1930 to 2014

home_teams = df_matches["Home Team Name"]
away_teams = df_matches["Away Team Name"]
unique_teams = pd.concat([home_teams, away_teams]).nunique() #sum of all the unique teams

df_matches["total_goals"] = df_matches["Home Team Goals"] + df_matches["Away Team Goals"]
average_goals = df_matches["total_goals"].mean() #average goals per match

print(total_matches)
print(missing_values)
print(year_range)
print(unique_teams)
print(average_goals)

4572
Year                    3720
Datetime                3720
Stage                   3720
Stadium                 3720
City                    3720
Home Team Name          3720
Home Team Goals         3720
Away Team Goals         3720
Away Team Name          3720
Win conditions          3720
Attendance              3722
Half-time Home Goals    3720
Half-time Away Goals    3720
Referee                 3720
Assistant 1             3720
Assistant 2             3720
RoundID                 3720
MatchID                 3720
Home Team Initials      3720
Away Team Initials      3720
dtype: int64
84.0
83
2.8333333333333335


In [29]:
import os
import requests
from dotenv import load_dotenv

load_dotenv("../.env")

api_key = os.getenv("FOOTBALL_API_KEY")
headers = {
    "X-Auth-Token": api_key
}
url = "https://api.football-data.org/v4/competitions/WC/matches"

response = requests.get(url, headers=headers)
data = response.json()

api_matches = pd.DataFrame(data.get("matches", []))

api_matches.head()

,area,competition,season,id,utcDate,status,matchday,stage,group,lastUpdated,homeTeam,awayTeam,score,odds,referees
0,"{'id': 2267, 'name': 'World', 'code': 'INT', '...","{'id': 2000, 'name': 'FIFA World Cup', 'code':...","{'id': 2398, 'startDate': '2026-06-11', 'endDa...",537327,2026-06-11T19:00:00Z,FINISHED,1.0,GROUP_STAGE,GROUP_A,2026-07-08T05:21:15Z,"{'id': 769, 'name': 'Mexico', 'shortName': 'Me...","{'id': 774, 'name': 'South Africa', 'shortName...","{'winner': 'HOME_TEAM', 'duration': 'REGULAR',...",{'msg': 'Activate Odds-Package in User-Panel t...,"[{'id': 11412, 'name': 'Wilton Sampaio', 'type..."
1,"{'id': 2267, 'name': 'World', 'code': 'INT', '...","{'id': 2000, 'name': 'FIFA World Cup', 'code':...","{'id': 2398, 'startDate': '2026-06-11', 'endDa...",537328,2026-06-12T02:00:00Z,FINISHED,1.0,GROUP_STAGE,GROUP_A,2026-07-08T05:21:15Z,"{'id': 772, 'name': 'South Korea', 'shortName'...","{'id': 798, 'name': 'Czechia', 'shortName': 'C...","{'winner': 'HOME_TEAM', 'duration': 'REGULAR',...",{'msg': 'Activate Odds-Package in User-Panel t...,"[{'id': 276653, 'name': 'Amin Omar', 'type': '..."
2,"{'id': 2267, 'name': 'World', 'code': 'INT', '...","{'id': 2000, 'name': 'FIFA World Cup', 'code':...","{'id': 2398, 'startDate': '2026-06-11', 'endDa...",537333,2026-06-12T19:00:00Z,FINISHED,1.0,GROUP_STAGE,GROUP_B,2026-07-08T05:21:15Z,"{'id': 828, 'name': 'Canada', 'shortName': 'Ca...","{'id': 1060, 'name': 'Bosnia-Herzegovina', 'sh...","{'winner': 'DRAW', 'duration': 'REGULAR', 'ful...",{'msg': 'Activate Odds-Package in User-Panel t...,"[{'id': 57301, 'name': 'Facundo Tello', 'type'..."
3,"{'id': 2267, 'name': 'World', 'code': 'INT', '...","{'id': 2000, 'name': 'FIFA World Cup', 'code':...","{'id': 2398, 'startDate': '2026-06-11', 'endDa...",537345,2026-06-13T01:00:00Z,FINISHED,1.0,GROUP_STAGE,GROUP_D,2026-07-08T05:21:15Z,"{'id': 771, 'name': 'United States', 'shortNam...","{'id': 761, 'name': 'Paraguay', 'shortName': '...","{'winner': 'HOME_TEAM', 'duration': 'REGULAR',...",{'msg': 'Activate Odds-Package in User-Panel t...,"[{'id': 43899, 'name': 'Danny Makkelie', 'type..."
4,"{'id': 2267, 'name': 'World', 'code': 'INT', '...","{'id': 2000, 'name': 'FIFA World Cup', 'code':...","{'id': 2398, 'startDate': '2026-06-11', 'endDa...",537334,2026-06-13T19:00:00Z,FINISHED,1.0,GROUP_STAGE,GROUP_B,2026-07-08T05:21:15Z,"{'id': 8030, 'name': 'Qatar', 'shortName': 'Qa...","{'id': 788, 'name': 'Switzerland', 'shortName'...","{'winner': 'DRAW', 'duration': 'REGULAR', 'ful...",{'msg': 'Activate Odds-Package in User-Panel t...,"[{'id': 190002, 'name': 'Said Martinez', 'type..."


1. The dataset contains 4572 matches and 21 World Cups.
2. The columns with missing values are: 3720 there no affects the data
3. The data covers the years from 1930 to 2014.
4. There are 83 unique national teams.
5. The historical average goals per match is 2.83.